spatialprofilingtoolbox/ondemand/request_handler.py

```diff
@@ -147,15 +147,27 @@ def _get_phenotype_pair_specification(self, groups):
 
     def _handle_squidpy_request(self, feature_class, groups):
         logger.debug(groups)
-        if (len(groups) < 3) or (len(groups) % 2 != 1):
-            return False
+        match feature_class:
+            case 'neighborhood enrichment':
+                if len(groups) != 4 + 1:
+                    return False
+            case 'co-occurrence':
+                if len(groups) != 4 + 1 + 1:
+                    return False
+            case 'ripley':
+                if len(groups) != 2 + 1:
+                    return False
```

Comment:
Why do neighborhood and ripley have different group lengths?

Reply:
The neighborhood enrichment metric depends on a focal phenotype, and another phenotype that is the one considered for being the "riches" that the former phenotype's cells are either "enriched in" or not. The Ripley statistic depends on a single point set, not a pair.